### PR TITLE
GH-678: Classification improvements

### LINKED ISSUE #670: Train & Test Loss scales
**Describe the bug**
I am training a text classification model as described in the tutorial. On plotting the training process, I see the the accuracy & F1 score plots converge and stabilize over 50 epochs. Whereas  The training and test loss have a wide gap though the stabilize.
On going over the loss.tsv it seemed that the test loss was not scaled as the training loss was.
On manually scaling the test loss, the new graph made sense.

On inspecting the flair code, the script https://github.com/zalandoresearch/flair/blob/master/flair/trainers/trainer.py Line 198, the issues seems to make sense as the training loss has been scaled.
flair_code

Can you please validate or clarify the same ?

**Expected behavior**
The graphs from Flair -
sample_imdb_training

The data used is a sample IMDB data set. 

The code was used from the Tutorial of Text Classification-
https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_7_TRAINING_A_MODEL.md

### PR DESCRIPTION
**Pull request by alanakbik**
This PR adds a number of small improvements and new features that were implemented in the course of investigating (but not yet solving) #678. 

#### FastText-style classification

FastText is a popular baseline for classifying text, based on learnable word embeddings that are mean-pooled. To allow Flair users to use this approach, we implemented a new word embedding type and adapted a document embedding: 

- **`OneHotEmbeddings`** - we add one-hot embeddings that are trainable on a downstream task. Instantiate by passing a corpus: 

```python
from flair.embeddings import OneHotEmbeddings
from flair.datasets import UD_ENGLISH

# a corpus
corpus = UD_ENGLISH()

# instantiate one-hot encoded word embeddings
embeddings = OneHotEmbeddings(corpus)
```

These embeddings can also be used to embed other fields (#398 #618). 

- Fine tuning options for **`DocumentPoolEmbeddings`** - we now allow users to specify a fine-tuning option before the pooling operation is executed in document pool embeddings. Options are 'none' (no fine-tuning), 'linear' (linear remapping of word embeddings), 'nonlinear' (nonlinear remapping of word embeddings). Nonlinear should be used together with `WordEmbeddings`, while None should be used with `OneHotEmbeddings` (not necessary since they are already learnt on data). So, to replicate FastText classification you can either do: 

```python
# instantiate one-hot encoded word embeddings
embeddings = OneHotEmbeddings(corpus)

# document pool embeddings
document_embeddings = DocumentPoolEmbeddings([embeddings], fine_tune_mode='none')
```

or 

```python
# instantiate pre-trained word embeddings
embeddings = WordEmbeddings('glove')

# document pool embeddings
document_embeddings = DocumentPoolEmbeddings([embeddings], fine_tune_mode='nonlinear')
```

#### Simplifications

Previously, users always had to specify whether their text classification problem was multi_label or not. Now, this is detected automatically if users do not specify. So now you can init like this: 

```python
# corpus
corpus = TREC_6()

# make label_dictionary
label_dictionary = corpus.make_label_dictionary()

# init text classifier
classifier = TextClassifier(document_embeddings, label_dictionary)
```

#### Others

- We added better module descriptions to embeddings and dropout so that more parameters get printed by default for models for better logging.
- We normalized the training loss across modules so that train / test loss are consistent. Closes #670 
- Improved speed of data loading for classification corpus
- Added TREC_6 classification corpus to `flair.datasets`

### CODE REVIEW

**Comment by kashif** [comment]
👍

**Comment by alanakbik** [comment]
:+1:
